slam-libraries/viam-cartographer/src/io/file_handler_test.cc

```diff
@@ -0,0 +1,183 @@
+    BOOST_TEST(filename_start.compare(path_to_dir + path_appendix) == 0);
+    // Extract timestamp
+    double filename_time = ReadTimeFromFilename(filename.substr(
+        filename.find(filename_prefix) + filename_prefix.length(),
+        filename.find(".pcd")));
+    // Check if timestamp is between start_time and end_time
+    BOOST_TEST((double)start_time <= filename_time);
+    BOOST_TEST(filename_time >= (double)end_time);
+}
+
+BOOST_AUTO_TEST_CASE(ListSortedFilesInDirectory_success) {
+    // Create a temp directory with a few files with timestamps
+    std::vector<std::string> files{
+        "rplidar_data_2022-01-01T01:00:00.0000Z.pcd",
+        "rplidar_data_2022-01-01T01:00:00.0001Z.pcd",
+        "rplidar_data_2022-01-01T01:00:00.0002Z.pcd",
+        "rplidar_data_2022-01-01T01:00:00.0003Z.pcd"};
+    // Create a unique path in the temp directory and add the files
+    boost::filesystem::path tmpdir = boost::filesystem::temp_directory_path() /
+                                     boost::filesystem::unique_path();
+    bool ok = boost::filesystem::create_directory(tmpdir);
+    if (!ok) {
+        throw std::runtime_error("could not create directory: " +
+                                 tmpdir.string());
+    }
+    for (std::string file : files) {
+        boost::filesystem::ofstream ofs(tmpdir / file);
+        ofs.close();
+    }
+    // List the sorted files in the directory
+    std::vector<std::string> listedFiles =
+        ListSortedFilesInDirectory(tmpdir.string());
+    // Check to make sure that the files in the directory are what we added and
+    // that they are in the same order
+    BOOST_TEST(files.size() == listedFiles.size());
+    for (int i = 0; i < files.size(); i++) {
+        BOOST_TEST(files.at(i).compare(listedFiles.at(i)));
+    }
+    // Close the file and remove the temporary directory and its contents
+    boost::filesystem::remove_all(tmpdir);
+}
+
+// test RemoveFile
+BOOST_AUTO_TEST_CASE(RemoveFile_success) {
+    // Create a temp directory with a few files with timestamps
+    std::vector<std::string> files{
+        "rplidar_data_2022-01-01T01:00:00.0000Z.pcd",
+        "rplidar_data_2022-01-01T01:00:00.0001Z.pcd",
+        "rplidar_data_2022-01-01T01:00:00.0002Z.pcd",
+        "rplidar_data_2022-01-01T01:00:00.0003Z.pcd"};
+    // Create a unique path in the temp directory and add the files
+    boost::filesystem::path tmpdir = boost::filesystem::temp_directory_path() /
+                                     boost::filesystem::unique_path();
+    bool ok = boost::filesystem::create_directory(tmpdir);
+    if (!ok) {
+        throw std::runtime_error("could not create directory: " +
+                                 tmpdir.string());
+    }
+    for (std::string file : files) {
+        boost::filesystem::ofstream ofs(tmpdir / file);
+        ofs.close();
+    }
+    // Remove a file
+    int success = RemoveFile(tmpdir.string() + "/" + files.at(1));
```

Comment:
[opt] could replace the one with a file_num variable that could be reused on line 91